wagtailmodelchoosers/client/sources/ModelSource.js

```diff
@@ -51,33 +52,35 @@ class ModelSource extends React.Component {
       }
       entityMutability = 'IMMUTABLE';
     }
-    const nextData = {
+    const entityData = {
       id,
       label,
       content_type,
     };
 
-    const nextState = DraftUtils.createEntity(
-      editorState,
+    const contentState = editorState.getCurrentContent();
+    const contentStateWithEntity = contentState.createEntity(
       type,
-      nextData,
-      nextData.label,
       entityMutability,
+      entityData,
+    );
+    const nextState = AtomicBlockUtils.insertAtomicBlock(
+      editorState,
+      contentStateWithEntity.getLastCreatedEntityKey(),
+      ' ',
```

Comment:
Atomic blocks are meant for block-level embeds, while I think most models this chooser is for would be considered inline. What you want is probably closer to https://github.com/springload/draftail/blob/9378913945f05395912a61f4b551e993c735c89f/examples/sources/LinkSource.js#L41-L51:

```js
        const contentStateWithEntity = contentState.createEntity(
            entityType.type,
            entityMutability,
            data,
        );
        const entityKey = contentStateWithEntity.getLastCreatedEntityKey();
        const nextState = RichUtils.toggleLink(
            editorState,
            editorState.getSelection(),
            entityKey,
        );
```

If it's indeed meant to be blocks, then the mutability selection above should be removed because atomic blocks cannot be mutable (since they don't use text).